internal/pkg/statemanager/state.go

```diff
@@ -67,3 +67,7 @@ type StateKey string
 func StateKeyGenerateFunc(t *configloader.Tool) StateKey {
 	return StateKey(fmt.Sprintf("%s_%s", t.Name, t.Plugin.Kind))
 }
+
+func GenStateKey(refParam []string) StateKey {
```

Comment:
"refParam" might be understandable (although not the best name) in the context of the pluginengine, but in the statemanager, it means nothing. The parameter's name can be improved.